### ISSUE TITLE
Restrict draggable element not to go out of viewport.

### ISSUE THREAD

**Issue by gokatz** [branu-ws/v-drag#13]
If we drag the element out of the viewport, we cannot bring them back to the viewport. Is there any way we can restrict dragging out of the viewport.?

**Comment by lmiller1990** [comment]
This is something we should support and would be fairly easy to do so.

Do you mean that any part of the boundary of the draggable object should not be allowed outside of the viewport?

I think this should probably be the default behavior.

**Comment by lmiller1990** [comment]
I will be working on and hopefully releasing this over the weekend @gokatz 👍

**Comment by gokatz** [comment]
Thank you so much @lmiller1990

**Comment by lmiller1990** [comment]
Added this @gokatz - please try it out. Not perfect but I think it can help you. Any feedback is welcome (not sure of your use case).

Install latest vers. with `npm install @branu-jp/v-drag@latest`

Added to the options https://github.com/branu-ws/v-drag#options , you can now use a `window-only` modifier.

```html
<div v-drag.window-only>
  This element cannot be dragged outside the window
</div>
```

**Comment by gokatz** [comment]
Thanks again for this @lmiller1990. Will try this and let you know. 

Just to clarify, my case is that the draggable element is a trigger to open a widget in a page. The user can drag the trigger anywhere on the page and while clicking the trigger, the widget will be toggled. I needed this option to restrict the user from dragging the container outside the window and if they do so, they cannot pull back the trigger.

**Comment by lmiller1990** [comment]
ok - this (should) work for that use case. Let me know!

**Comment by lmiller1990** [comment]
@gokatz was this okay?

**Comment by gokatz** [comment]
Yeah. Perfectly works for my case @lmiller1990. one little thing that I wish to clarify.
In this sandbox, https://codesandbox.io/s/r4nr1nq0vp, when the draggable element is dragged to the edge of the viewport, scrollbars began to appear. Am I missing something? 

<img width="1438" alt="screen shot 2018-08-24 at 2 04 51 pm" src="https://user-images.githubusercontent.com/15158776/44574469-1fd17d80-a7a7-11e8-8791-468d9a86c6b9.png">

**Comment by lmiller1990** [comment]
Huh, it happened on the right and bottom areas. Maybe the check if not triggering fast enough, and before x/y location is constrained, the browser moves the viewport to accommodate. 

I'll be improving this on the weekend, for sure. Glad this is at least working for your use case, thanks for the feedback.

**Comment by gokatz** [comment]
Anytime. Thanks for your time 🙂
